Docs: Update and proofread `library/venv.rst` (#124121)
python/cpython: Doc/library/venv.rst

  are by default isolated from software in other virtual environments and Python
  interpreters and libraries installed in the operating system.

> * Contained in a directory, conventionally either named ``.venv`` or ``venv`` in

Reviewer: ```suggestion
* Contained in a directory, conventionally named ``.venv`` or ``venv`` in
```

The (incumbent) "either" here seems a bit ambiguous and confusing (it confused be for a bit), as it could be referring to either (heh) of the names or of the locations, and I don't think it really contribute to clarity. I considered moving it or substituting another word, but to me removing it seemed like the simplest solution.

Creating virtual environments
-----------------------------

:ref:`Virtual environments <venv-def>` are created by executing the ``venv``
module:

.. code-block:: shell

    python -m venv /path/to/new/virtual/environment

This creates the target directory (and any parent
directories that don't already exist) and places a ``pyvenv.cfg`` file in it
with a ``home`` key pointing to the Python installation from which the command
was run.  It also creates
a ``bin`` (or ``Scripts`` on Windows) subdirectory containing a copy/symlink
of the Python binary/binaries (as appropriate for the platform or arguments
used at environment creation time). It also creates an (initially empty)
``lib/pythonX.Y/site-packages`` subdirectory (on Windows, this is
``Lib\site-packages``). If an existing directory is specified, it will be

> re-used.

Reviewer 2: Re-wrap and minor rewording
```suggestion
This creates the target directory (including parent directories as needed)
and places a :file:`pyvenv.cfg` file in it with a ``home`` key
pointing to the Python installation from which the command was run.
It also creates a :file:`bin` (or :file:`Scripts` on Windows) subdirectory
containing a copy or symlink of the Python executable
(as appropriate for the platform or arguments used at environment creation time).
It also creates a :file:`lib/pythonX.Y/site-packages` subdirectory
(on Windows, this is :file:`Lib\site-packages`).
If an existing directory is specified, it will be re-used.
```

                           environment (pip is bootstrapped by default)
     --prompt PROMPT       Provides an alternative prompt prefix for this
                           environment.
     --upgrade-deps        Upgrade core dependencies (pip) to the latest
                           version in PyPI
     --without-scm-ignore-files
                           Skips adding SCM ignore files to the environment
                           directory (Git is supported by default).

   Once an environment has been created, you may wish to activate it, e.g. by
   sourcing an activate script in its bin directory.

.. versionchanged:: 3.4
   Installs pip by default, added the ``--without-pip``  and ``--copies``
   options.

.. versionchanged:: 3.4
   In earlier versions, if the target directory already existed, an error was
   raised, unless the ``--clear`` or ``--upgrade`` option was provided.

.. versionchanged:: 3.9
   Add ``--upgrade-deps`` option to upgrade pip + setuptools to the latest on PyPI.

.. versionchanged:: 3.12

   ``setuptools`` is no longer a core venv dependency.

.. versionchanged:: 3.13

   Added the ``--without-scm-ignore-files`` option.

>    

Reviewer 2: To fix the CI failure
```suggestion

```

.. versionchanged:: 3.13
   ``venv`` now creates a :file:`.gitignore` file for Git by default.

.. note::
   While symlinks are supported on Windows, they are not recommended. Of
   particular note is that double-clicking ``python.exe`` in File Explorer
   will resolve the symlink eagerly and ignore the virtual environment.

.. note::
   On Microsoft Windows, it may be required to enable the ``Activate.ps1``
   script by setting the execution policy for the user. You can do this by
   issuing the following PowerShell command:

   .. code-block:: powershell

      PS C:\> Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser

   See `About Execution Policies
   <https://go.microsoft.com/fwlink/?LinkID=135170>`_
   for more information.

> The created ``pyvenv.cfg`` file also includes the

Reviewer 2: ```suggestion
The created :file:`pyvenv.cfg` file also includes the
```

Creating virtual environments
-----------------------------

:ref:`Virtual environments <venv-def>` are created by executing the ``venv``
module:

.. code-block:: shell

    python -m venv /path/to/new/virtual/environment

This creates the target directory (and any parent
directories that don't already exist) and places a ``pyvenv.cfg`` file in it
with a ``home`` key pointing to the Python installation from which the command

> was run (a common name for the target directory is ``.venv``).  It also creates

Reviewer 2: Duplicates the above naming guidance
```suggestion
was run.  It also creates
```

a ``bin`` (or ``Scripts`` on Windows) subdirectory containing a copy/symlink
of the Python binary/binaries (as appropriate for the platform or arguments
used at environment creation time). It also creates an (initially empty)
``lib/pythonX.Y/site-packages`` subdirectory (on Windows, this is
``Lib\site-packages``). If an existing directory is specified, it will be
re-used.

.. versionchanged:: 3.5
   The use of ``venv`` is now recommended for creating virtual environments.

.. deprecated-removed:: 3.6 3.8
   :program:`pyvenv` was the recommended tool for creating virtual environments
   for Python 3.3 and 3.4, and replaced in 3.5 by executing ``venv`` directly.

.. highlight:: none

> On Windows, invoke the ``venv`` command as follows::
> 
>     c:\>Python35\python -m venv c:\path\to\myenv
> 
> Alternatively, if you configured the ``PATH`` and ``PATHEXT`` variables for
> your :ref:`Python installation <using-on-windows>`::
> 
>     c:\>python -m venv c:\path\to\myenv

Reviewer 2: We should present the most common scenario (which isn't Python 3.5 built from source)
```suggestion
On Windows, invoke the ``venv`` command as follows:

.. code-block:: ps1con

   PS> python -m venv C:\path\to\new\virtual\environment
```

     --without-pip         Skips installing or upgrading pip in the virtual
                           environment (pip is bootstrapped by default)
     --prompt PROMPT       Provides an alternative prompt prefix for this
                           environment.
     --upgrade-deps        Upgrade core dependencies (pip) to the latest
                           version in PyPI
     --without-scm-ignore-files
                           Skips adding SCM ignore files to the environment
                           directory (Git is supported by default).

   Once an environment has been created, you may wish to activate it, e.g. by
   sourcing an activate script in its bin directory.

.. versionchanged:: 3.4
   Installs pip by default, added the ``--without-pip``  and ``--copies``
   options.

.. versionchanged:: 3.4
   In earlier versions, if the target directory already existed, an error was
   raised, unless the ``--clear`` or ``--upgrade`` option was provided.

.. versionchanged:: 3.9
   Add ``--upgrade-deps`` option to upgrade pip + setuptools to the latest on PyPI.

.. versionchanged:: 3.12

   ``setuptools`` is no longer a core venv dependency.

.. versionchanged:: 3.13

>    ``--without-scm-ignore-files`` was added along with creating an ignore file
>    for Git by default.

Reviewer 2: ```suggestion
   Added the ``--without-scm-ignore-files`` option.
   
.. versionchanged:: 3.13
   ``venv`` now creates a :file:`.gitignore` file for the Git VCS by default.
```

Author: We've already introduced Git, how about "for Git by default" without needing to refer to it as a VCS?

```suggestion
   Added the ``--without-scm-ignore-files`` option.
   
.. versionchanged:: 3.13
   ``venv`` now creates a :file:`.gitignore` file for Git by default.
```

In order to achieve this, scripts installed into virtual environments have
a "shebang" line which points to the environment's Python interpreter,

> that is :samp:`#!/{<path-to-venv>}/bin/python`.

Reviewer: ```suggestion
:samp:`#!/{<path-to-venv>}/bin/python`.
```

Wouldn't it be even simpler without the extra words; that is, just stating the Python interpreter path directly?

    .. method:: install_scripts(context, path)

        *path* is the path to a directory that should contain subdirectories

>         ``common``, ``posix``, ``nt``, each containing scripts destined for the
>         bin directory in the environment.  The contents of ``common`` and the

Reviewer: ```suggestion
        ``common``, ``posix``, ``nt``; each containing scripts destined for the
        ``bin`` directory in the environment.  The contents of ``common`` and the
```

* Use literals for ``bin`` as well, since it is a directory like the others
* Use semicolon to more clearly separate the items in the list from the next clause, especially when eliding "and"

    .. method:: create_git_ignore_file(context)

       Creates a ``.gitignore`` file within the virtual environment that causes

>        the entire directory to be ignored by the Git source control manager.

Reviewer: IMO, most folks seem to call it a VCS instead of SCM these days, but I digress

Author: In the docs, we have:

* 2 "version control system" + 3 VCS
* 1 "source control manager" + 5 SCM

Too close to be decisive.

I don't mind, feel free add a a suggest if you want to change and I'll accept it :)
